### ISSUE TITLE
Lecture 4 challenges

### ISSUE THREAD

**Issue by daniellekarakas** [BME1478H/Fall2022class#400]

**Comment by daniellekarakas** [comment]
Hi,

I am trying to complete the lecture 4 challenges, but am getting an error message despite copying the code similarly to the lecture mark down solutions. I have attached a screenshot of the issue, as well as my code below. 

`class Cat:
    def _init_(self, name, eye_color, hair_type, behaviour, price): 
        self.name=name
        self.eye_color=eye_color
        self.hair_type=hair_type 
        self.behaviour=behaviour
        self.price=price
        
    def description(self):
        desc_str=f"{self.name} is a {self.eye_color}-eyed {self.behaviour} cat with {self.hair_type} worth {self.price}."
        return desc_str

C1=Cat()
C1.name= 'Milly'
C1.eye_color='green'
C1.hair_type='short'
C1.behaviour='calm'
C1.price='$100'

C1.description()

'Milly is a green-eyed calm cat with short worth $100.'`

In this case, I get the expected output. However, when I try and follow it with:

`C2=Cat('Birman','blue eyes', 'cuddly', 'long hair', '$600')`

I get the following error message:

TypeError: Cat() takes no arguments

I am not sure why this is happening as I am mirroring the same format used in lecture, just with a different object variable? I have also attached a screenshot. 
<img width="1019" alt="Screen Shot 2022-02-07 at 10 30 53 PM" src="https://user-images.githubusercontent.com/97863667/152913347-8ef153ee-6fe7-487a-9b9c-07757c16543a.png">
#

**Comment by RayNele** [comment]
Hey Danielle, 

you're actually supposed to replace the issue template with the  your question so that the first post is your question (the third last line of the template).

Now to actually answer your question, you need two underscores before and after init like so: ```__init__```

**Comment by daniellekarakas** [comment]
Hey sorry about the formatting, but I do have two underscores.

**Comment by RayNele** [comment]
It's hard to keep track of what your problem is since you edited the previous comment. Let's just keep posting new comments instead so we can build up on the solution and other people can see how we arrived at the solution. 

The first time you asked the question, your ```__init__``` did not have two underscores before and after so the function wasn't working as intended. So you created a class that did not need 5 arguments. That's why you could make a C1 object in that class using ```C1=Cat()```


Now that you've fixed the double underscore problem, your class Cat NEEDS to have those 5 variables. You can set optional arguments, but python assumes that if you do not specify them as optional, then they are required (hence the error message). 

Basically before you broke the part that said it needed 5 arguments, that's why it worked without 5 arguments. 

You fixed the part with 5 arguments, so now you need 5 arguments.
